network_test.go

```diff
@@ -328,17 +350,27 @@ func testSwarmNetwork(t *testing.T, o *testSwarmNetworkOptions, steps ...testSwa
 				})
 			}
 
-			if *waitKademlia {
-				if _, err := sim.WaitTillHealthy(ctx); err != nil {
-					return err
+			for syncing := true; syncing; {
+				syncing = false
+				for _, id := range nodeIDs {
+					if sim.MustNodeItem(id, bucketKeyInspector).(*api.Inspector).IsPullSyncing() {
+						syncing = true
```

Comment:
I think that after `syncing = true` a break can go.

Reply:
you mean `continue`. if `syncing == true` then the loop keeps on going till the syncing stops, but in any case, i'd rather have it hit the `Sleep` before checking all nodes again